web/src/composebox_typeahead.ts

```diff
@@ -642,27 +736,84 @@ export function get_sorted_filtered_items(query, input_element) {
 
     const big_results = get_candidates(query, input_element);
 
-    if (!big_results) {
-        return false;
+    if (big_results === false) {
+        return [];
```

Comment:
`source` is expecting a list so `false` doesn't make sense here. It was working before because the code that calls `source` just does falsey checks

```js
        const items = this.source(this.query, this.input_element);

        if (!items && this.shown) {
            this.hide();
        }
        return items ? this.process(items) : this;
```